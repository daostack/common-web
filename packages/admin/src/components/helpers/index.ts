export { ClientOnly } from './ClientOnly';
export { StatusIcon } from './StatusIcon';