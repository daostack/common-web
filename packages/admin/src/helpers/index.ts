export * from './apolloHelper';

export interface IPaginate {
  take: number;
  skip: number;
}

export const DefaultPaginate: IPaginate = {
  skip: 0,
  take: 10
};