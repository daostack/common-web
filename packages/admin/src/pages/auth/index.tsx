import { NextPage } from 'next';
import React from 'react';
import { Button, Description, Grid, Text, Tooltip, useMediaQuery, useTheme, useToasts } from '@geist-ui/react';
import { CheckInCircleFill } from '@geist-ui/react-icons';
import firebase from 'firebase/app';
import { AuthEmission } from '@react-firebase/auth/dist/types';
import { useRouter } from 'next/router';

interface IAuthenticatePageProps {
  auth: AuthEmission;
}

const AuthenticationPage: NextPage<IAuthenticatePageProps> = ({ auth }) => {
  const theme = useTheme();
  const router = useRouter();
  const isLarge = useMediaQuery('md', { match: 'up' });
  const [toasts, setToasts] = useToasts();

  React.useEffect(() => {
    if(auth.isSignedIn) {
      router.push('/dashboard');
    }
  }, []);

  const singIn = (provider: firebase.auth.AuthProvider) => {
    return async () => {
      try {
        const signInResult = await firebase.auth()
          .signInWithPopup(provider);

        // Check permissions
        console.log(signInResult)

        setToasts({
          type: 'success',
          text: 'Successfully authenticated'
        })
      } catch (e) {
        if (e.code === 'auth/popup-closed-by-user') {
          setToasts({
            type: 'error',
            text: e.message
          });
        } else {
          setToasts({
            type: 'error',
            text: 'Something went wrong'
          });

          console.error(e);
        }
      }

    };
  };

  return (

    <Grid.Container style={{ height: '100vh' }}>
      <Grid md={12} style={{
        background: theme.palette.accents_1,
        display: 'flex',
        justifyContent: 'flex-end',
        alignItems: 'center'
      }}>
        <div style={{ width: '25vw', marginRight: '5vw' }}>
          <Text h2>Common Admin</Text>

          <div style={{ display: 'flex', marginTop: 30 }}>
            <div style={{ marginRight: 10 }}>
              <CheckInCircleFill color={theme.palette.success}/>
            </div>

            <Description
              title="Instant access to statistic"
              content="See statistics that count in almost real time from the common count to the open proposals and much more"
            />
          </div>

          <div style={{ display: 'flex', marginTop: 30 }}>
            <div style={{ marginRight: 10 }}>
              <CheckInCircleFill color={theme.palette.success}/>
            </div>

            <Description
              title="Users at a glance"
              content="See all the users, see their commons and proposals, and help them with their payments and subscriptions"
            />
          </div>

          <div style={{ display: 'flex', marginTop: 30 }}>
            <div style={{ marginRight: 10 }}>
              <CheckInCircleFill color={theme.palette.success}/>
            </div>

            <Description
              title="Commons management"
              content="See all the commons, see their members and proposals, remove common member upon request and more"
            />
          </div>
        </div>
      </Grid>

      <Grid sm={24} md={12} style={{
        background: theme.palette.background,
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'center',
        padding: '5vw'
      }}>
        <div style={{ ...(isLarge && { width: '25vw' }) }}>
          <Text h2>Sign in to continue</Text>

          <div style={{ margin: '10px 0' }}>
            <Button
              shadow
              onClick={singIn(new firebase.auth.GoogleAuthProvider())}
              style={{ width: '100%', margin: '5px 0' }}
            >
              Continue with Google
            </Button>

            <Tooltip text="Sign in with Apple is yet to be implemented" enterDelay={1000} style={{ width: '100%' }}>
              <Button style={{ width: '100%', margin: '5px 0', cursor: 'not-allowed' }} shadow type="secondary">
                Continue with Apple
              </Button>
            </Tooltip>
          </div>

          <Text>
            By clicking continue, you agree to our Terms of Service and Privacy Policy.
          </Text>
        </div>
      </Grid>
    </Grid.Container>

  );
};

export default AuthenticationPage;