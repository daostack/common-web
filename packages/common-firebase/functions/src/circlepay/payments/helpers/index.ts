export { isFinalized, isSuccessful } from './statusHelper';

export { updatePayments } from './converter';

export { feesHelper } from './feesHelper';
export { failureHelper } from './failureHelper';