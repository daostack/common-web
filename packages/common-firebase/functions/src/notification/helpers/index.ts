export const getFundingRequestAcceptedTemplate = (country: string, amount: number) => {
  if (amount) {
    return !country
      ? 'userFundingRequestAcceptedUnknown'
      : (country === 'IL'
        ? 'userFundingRequestAcceptedIsraeli'
        : 'userFundingRequestAcceptedForeign');
  }
  return 'userFundingRequestAcceptedZeroAmount';
};