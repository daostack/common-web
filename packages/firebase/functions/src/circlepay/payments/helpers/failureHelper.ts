import { ICirclePayment } from '../../types';

import { IPaymentFailureReason, PaymentFailureResponseCodes } from '../types';

const failureCodeDescription: {
  [key in PaymentFailureResponseCodes]: string;
} = {
  card_account_ineligible: 'Ineligible account associated with card',
  card_invalid: 'Invalid card number',
  card_limit_violated: 'Exceeded amount or frequency limits',
  card_not_honored: 'Contact card issuer to query why payment failed',
  credit_card_not_allowed: 'Issuer did not support using a credit card for payment',
  payment_denied: 'Payment denied by Circle Risk Service or card processor risk controls',
  payment_failed: 'Payment failed due to unspecified error',
  payment_fraud_detected: 'Payment suspected of being associated with fraud',
  payment_not_funded: 'Insufficient funds in account to fund payment',
  payment_not_supported_by_issuer: 'Issuer did not support the payment',
  payment_stopped_by_issuer: 'A stop has been placed on the payment or card'

};

const getFailureDescription = (responseCode: PaymentFailureResponseCodes = 'payment_failed'): string =>
  failureCodeDescription[responseCode];

export const processFailedPayment = (payment: ICirclePayment): IPaymentFailureReason => {
  if (payment.data.status !== 'failed') {
    logger.warn('Trying to process failed payment that is not actually failed', {
      payment
    });

    return null;
  }

  const failureReason: IPaymentFailureReason = {
    errorCode: payment.data.errorCode || 'payment_failed',
    errorDescription: getFailureDescription(payment.data.errorCode)
  };

  logger.debug(`Processed failure reason for payment}`, {
    failureReasonFromCircle: payment.data.errorCode,
    processedFailureReason: failureReason,
    payment
  });

  return failureReason;
};

export const failureHelper = {
  processFailedPayment
};