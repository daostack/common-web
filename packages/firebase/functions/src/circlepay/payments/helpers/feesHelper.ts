import { ICirclePayment } from '../../types';
import { IPaymentFees } from '../types';

const defaultFees: IPaymentFees = {
  amount: 0,
  currency: 'USD'
};

export const processCircleFee = (circlePayment: ICirclePayment): IPaymentFees => {
  const { data: payment } = circlePayment;

  if (!payment.fees) {
    logger.warn('Trying to process payment fees, but there is no fees object on the payment', {
      payment
    });

    return defaultFees;
  }

  const fees: IPaymentFees = {
    // Circle returns the fees in dollars. Convert it to cents,
    amount: Number(payment.fees.amount) * 100,
    currency: payment.fees.currency
  };

  logger.debug('Converted circle fees', {
    circleFees: payment.fees,
    convertedFees: fees,
    circlePaymentId: circlePayment.data.id
  });

  return fees;
};

export const feesHelper = {
  processCircleFee
}