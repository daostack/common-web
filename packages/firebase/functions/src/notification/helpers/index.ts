import { env } from '../../constants';

export const getFundingRequestAcceptedTemplate = (country: string, amount: number): string => {
  if (amount) {
    return !country
      ? 'userFundingRequestAcceptedUnknown'
      : (country === 'IL'
        ? 'userFundingRequestAcceptedIsraeli'
        : 'userFundingRequestAcceptedForeign');
  }
  return 'userFundingRequestAcceptedZeroAmount';
};

//indication the email was sent from staging
export const testFlag = (): string => env.environment === 'staging' || env.environment === 'dev' ? '[TEST] ' : '';
