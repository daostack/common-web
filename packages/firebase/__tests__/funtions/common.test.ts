// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import firebaseFunctionsTests from 'firebase-functions-test';

import { getTestAuthToken } from '../helpers/auth';
import { commonApp } from '../helpers/supertests';

const test = firebaseFunctionsTests({
  projectId: 'common-tests'
});

const invalidCommonCreationPayload = {
  name: 'Hello, I\'m a test',
  description: 'I should not be created tho as I\'m incomplete'
};

const validCommonCreationPayload = {
  name: 'Common Test',
  image: 'https://llandscapes-10674.kxcdn.com/wp-content/uploads/2019/07/lighting.jpg',
  byline: 'basically im long byline',
  description: 'hey there, am i descriptive',
  contributionType: 'one-time',
  contributionAmount: 6500,
  zeroContribution: true,
};

describe('Common Related Cloud Functions', () => {
  afterAll(async () => {
    await test.cleanup();
  });

  it('should work', () => {
    expect(true).toBeTruthy();
  });

  it('should be healthy', async () => {
    const authToken = await getTestAuthToken('test-user');

    const response = await commonApp
      .get('/health')
      .set({
        Authorization: authToken
      });


    expect(response.status).toBe(200);
    expect(response.body.healthy).toBeTruthy();
  });

  describe('Common Creation', () => {
    it('should not allow unauthorized requests', async () => {
      const response = await commonApp
        .post('/create');

      expect(response.status).toBe(401);
    });

    it('should fail validation on invalid input', async () => {
      // Setup
      const authToken = await getTestAuthToken('test-user');

      const invalidContributionType = {
        ...validCommonCreationPayload,
        contributionType: 'something wrong'
      };

      // Act
      const response = await commonApp
        .post('/create')
        .set({
          Authorization: authToken
        })
        .send(invalidCommonCreationPayload);

      const invalidContributionResponse = await commonApp
        .post('/create')
        .set({
          Authorization: authToken
        })
        .send(invalidContributionType);

      // Assert
      expect(response.status).toBe(422);

      expect(response.body.errorCode).toBeDefined();
      expect(response.body.errorCode).toBe('ValidationError');

      expect(response.body.data.errors).toMatchSnapshot();

      expect(
        invalidContributionResponse.body.data.errors
          .some(x => x === 'contributionType must be one of the following values: one-time, monthly')
      ).toBeTruthy();
    });

    it('should create common successfully given valid data', async () => {
      // Setup
      const authToken = await getTestAuthToken('test-user');

      // Act
      const response = await commonApp
        .post('/create')
        .set({
          Authorization: authToken
        })
        .send(validCommonCreationPayload);

      // Assert
      expect(response.status).toBe(200);

      expect(response.body.register).toBe('na');
      expect(response.body.members.length).toBe(1);
      expect(response.body.metadata.founderId).toBe('test-user');
    });
  });
});