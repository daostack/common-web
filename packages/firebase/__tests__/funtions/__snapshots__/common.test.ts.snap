// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Related Cloud Functions Common Creation should fail validation on invalid input 1`] = `
Array [
  "You must provide a valid URL",
  "image is a required field",
  "fundingGoalDeadline is a required field",
  "contributionAmount is a required field",
]
`;
